pkg/apiserver/handlers.go

```diff
@@ -188,11 +190,11 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	done := make(chan struct{}, 1)
+	done := make(chan struct{})
 	tw := newTimeoutWriter(w)
 	go func() {
 		t.handler.ServeHTTP(tw, r)
```

Comment:
> We can worry about that separately, it doesn't need to block this PR :)

Ha... That is also my point:https://github.com/kubernetes/kubernetes/pull/29594#discussion_r73968141

I created another issue for this: see https://github.com/kubernetes/kubernetes/issues/30305.